### ISSUE TITLE
Invalid array accepted in Dockerfile

### ISSUE THREAD

**Issue by pslacerda** [moby/moby#36348]
The Dockerfile that follows is currently valid.

```Dockerfile
FROM python:3
RUN ["ls"],
```

But maybe it should not. I expect to the trailing `,` be rejected.

Is that the correct behaviour?

**Comment by rcjsuen** [comment]
I don't think this is a bug even though the behaviour seems a little odd. Please see #32074.
